test/parallel/test-dns-promises-lookupService.js

```diff
@@ -0,0 +1,17 @@
+'use strict';
+
+const common = require('../common');
+
+const assert = require('assert');
+const dnsPromises = require('dns').promises;
+
+dnsPromises.lookupService('127.0.0.1', 22).then(common.mustCall((result) => {
+  assert.strictEqual(result.service, 'ssh');
+  assert.strictEqual(typeof result.hostname, 'string');
+  assert.notStrictEqual(result.hostname.length, 0);
+}));
+
+assert.rejects(
+  () => dnsPromises.lookupService('0.0.0.0', 22),
+  { code: /^(?:ENOTFOUND|EAI_AGAIN)$/ }
+);
```

Comment:
This doesn’t seem to reject on Windows – maybe we need to skip this part of the test there?

Reply:
Yeah, it works on some Windows machines in CI but not on others. I want to look a bit more closely at it before skipping....

Reply:
Oh wait, it failed on all....

Reply:
Running this locally on my Windows 10 I get...

```js
> dnsPromises.lookupService('0.0.0.0', 22).then(console.log).catch(console.log)
Promise { <pending> }
> { hostname: 'DESKTOP-SGNU281', service: 'ssh' }
```

Reply:
Thanks! That's unfortunate (for me, anyway) and (at least as I understand the meaning of `0.0.0.0`) not spec-compliant (although I could be wrong). Then again, I think lots of web browsers "correct" `0.0.0.0` to `127.0.0.1` or some other variant of `localhost`, so...OK. 

I'll try an address from the RFC 5737 test range and hope it works. If not, and if neither of the other two test ranges work, I guess skipping on Windows will do.

Reply:
Maybe use one of the reserved IP addresses (e.g. https://tools.ietf.org/html/rfc2606, https://tools.ietf.org/html/rfc5737)?

Reply:
> I'll try an address from the RFC 5737 test range and hope it works

That worked! And confirmed that it also works with my network turned off.